comtypes/test/test_dispifc_safearrays.py

```diff
@@ -0,0 +1,88 @@
+# coding: utf-8
+
+import unittest
+import comtypes
+
+from comtypes import CLSCTX_LOCAL_SERVER
+from comtypes.client import CreateObject, GetModule
+from ctypes import byref, pointer, c_double
+
+ComtypesCppTestSrvLib_GUID = "{07D2AEE5-1DF8-4D2C-953A-554ADFD25F99}"
+
+try:
+    GetModule((ComtypesCppTestSrvLib_GUID, 1, 0, 0))
+    import comtypes.gen.ComtypesCppTestSrvLib as ComtypesCppTestSrvLib
+
+    IMPORT_FAILED = False
+except (ImportError, OSError):
+    IMPORT_FAILED = True
+
+
+@unittest.skipIf(IMPORT_FAILED, "This depends on the out of process COM-server.")
+class Test_DispMethods(unittest.TestCase):
+    """Test dispmethods with safearray and safearray pointer parameters."""
+
+    UNPACKED_ZERO_VALS = tuple(0.0 for _ in range(10))
+    UNPACKED_CONSECUTIVE_VALS = tuple(float(i) for i in range(10))
+
+    def _create_dispifc(self) -> "ComtypesCppTestSrvLib.IDispSafearrayParamTest":
+        # Explicitely ask for the dispinterface of the component.
+        return CreateObject(
+            "Comtypes.DispSafearrayParamTest",
+            clsctx=CLSCTX_LOCAL_SERVER,
+            interface=ComtypesCppTestSrvLib.IDispSafearrayParamTest,
+        )
+
+    def _create_zero_array(self) -> comtypes.safearray._midlSAFEARRAY:
+        return comtypes.safearray._midlSAFEARRAY(c_double).create(
+            [c_double() for _ in range(10)]
+        )
+
+    def _create_consecutive_array(self) -> comtypes.safearray._midlSAFEARRAY:
+        return comtypes.safearray._midlSAFEARRAY(c_double).create(
+            [c_double(i) for i in range(10)]
+        )
```

Comment:
Indeed, I suggested annotating the return value with `comtypes.safearray._midlSAFEARRAY`.

However, I noticed that `_midlSAFEARRAY` is not a type, but a function that returns a pointer to a `tagSAFEARRAY` derived type.
It is difficult to represent such type information with the current Python type system.

Given that the method name implies it returns an "array", I will remove the return type annotation.

As for the appropriate symbol for the return value of `_midlSAFEARRAY`, I plan to address it in a separate topic.

```suggestion
    def _create_zero_array(self):
        return comtypes.safearray._midlSAFEARRAY(c_double).create(
            [c_double() for _ in range(10)]
        )

    def _create_consecutive_array(self):
        return comtypes.safearray._midlSAFEARRAY(c_double).create(
            [c_double(i) for i in range(10)]
        )
```